openlabcmd/openlabcmd/plugins/nodepool/orphanresource.py

```diff
@@ -0,0 +1,253 @@
+import datetime
+import iso8601
+import json
+import requests
+import subprocess
+import yaml
+
+from openlabcmd.plugins.base import Plugin
+
+SERVER_WHITE_LIST = [
+    '%s-openlab-zuul', '%s-openlab-nodepool', '%s-openlab-zookeeper']
+OPENSTACK_CLOUD_CFG_PATH = '/etc/openstack/clouds.yaml'
+# DAYs
+RESOURCE_TIMEOUT = 1
+
+
+class OrphanResourcePlugin(Plugin):
+
+    # This Plugin is only for checking, as it's a dangerous action for each.
+    ptype = 'nodepool'
+    name = 'OrphanResource'
+
+    def _get_endpoint_and_token(self):
```

Comment:
question: Not sure why we could not use the cli to get the token and endpoint?

Reply:
use openstacksdk directly is easier and simpler.